lib/controllers/user.controller.js

```diff
@@ -0,0 +1,62 @@
+import UserService from '../services/user.service';
+import CommonValidator from '../validators/common.validator';
+import UserValidator from '../validators/user.validator';
+
+export default class UserController {
+    constructor() {
+        this.userService = new UserService();
+        this.commonValidator = new CommonValidator();
+        this.userValidator = new UserValidator();
+    }
+
+    find(req, res, next, user_id) {
+        req.user = this.userService.find(user_id);
+        if (req.user) {
+            return next();
+        }
+
+        res.status(404).json({ message: `User with id ${user_id} not found` });
+    }
+
+    getUser(req, res) {
+        res.json(req.user);
+    }
+
+    getUsers(req, res) {
+        const { loginSubstring, limit } = req.query;
+
+        if (loginSubstring && limit) {
+            if (this.commonValidator.isNotPositive(limit)) {
```

Comment:
This validation can be used as middleware

Reply:
Done